### ISSUE TITLE
the name of lambda method that compiled by gradle7.3 is different from maven

### ISSUE THREAD

**Issue by wangxu163** [gradle/gradle#20949]
<img width="1169" alt="image" src="https://user-images.githubusercontent.com/30898306/172341824-7be4e93b-b2df-4da0-89d5-dc02053f0c77.png">

**Comment by wangxu163** [comment]
the left class file is compiled by gradle, the right class file is compiled by maven,both of them are generated by javap -v.
i want to know why are they different and does the difference affect the code running?

**Comment by h0tk3y** [comment]
Thank you for your interest in Gradle!

Your issue appears to be a question about how to use Gradle. Your question might be better answered on our forums or StackOverflow. We have several resources on [help.gradle.org](https://help.gradle.org/) that you should check out first.

If you think you have come across a Gradle bug, please file a new issue describing the problem and how to reproduce it.  

If you think Gradle is missing a feature, please file a new issue describing your use case for the feature.

---

In short, this difference should not actually affect the semantics of the compiled code.
